controllers/integrationpipeline/integrationpipeline_adapter.go

```diff
@@ -333,6 +370,18 @@ func (a *Adapter) determineIfAllIntegrationPipelinesPassed(integrationPipelineRu
 			// prevent mess in logs
 			pipelineRunOutcome.LogResults(a.logger.Logger)
 		}
+
+		// Remove the finalizer from this Integration PipelineRun
```

Comment:
I'm thinking it would make more sense to have the finalizer removed at the end of [EnsureStatusReportedInSnapshot](https://github.com/redhat-appstudio/integration-service/blob/2e6db82e809d22589bf20b69c1382580d7e66f4a/controllers/integrationpipeline/integrationpipeline_adapter.go#L194) since that is the crucial point where we record the pipeline testing outcome.